extension/src/in-page/index.ts

```diff
@@ -1,223 +1,108 @@
 import Debug from "debug";
-import { Direction, Message, MessageKind } from "../messages";
+import { AsyncReturnType } from "type-fest";
+import { v4 } from "uuid";
+import { RequestMessage, ResponseMessage } from "../contentScript";
 import { Address, CreateSwapPayload, LoanRequestPayload, Tx, Txid, WalletStatus } from "../models";
 
 Debug.enable("*");
 const debug = Debug("inpage");
 
 export default class WavesProvider {
     public async walletStatus(): Promise<WalletStatus> {
-        debug("Requesting wallet status");
-        let promise = new Promise<WalletStatus>((resolve, reject) => {
-            let listener = async function(event: MessageEvent<Message<WalletStatus>>) {
-                if (
-                    event.data.direction === Direction.ToPage
-                    && event.data.kind === MessageKind.WalletStatusResponse
-                ) {
-                    if (event.data.error) {
-                        reject(event.data.error);
-                    } else {
-                        debug(`Received wallet status: ${JSON.stringify(event.data)}`);
-
-                        window.removeEventListener("message", listener);
-                        resolve(event.data.payload);
-                    }
-                }
-            };
-            window.addEventListener("message", listener);
-        });
-        window.postMessage({
-            kind: MessageKind.WalletStatusRequest,
-            direction: Direction.ToBackground,
-        }, "*");
-        return promise;
+        return invokeContentScript("walletStatus", []);
```

Comment:
This can be implemented using a JS proxy in the future if it gets too cumbersome. Can easily happen in a follow-up!